pollux/lib/core/src/main/scala/io/iohk/atala/pollux/core/model/schema/type/CredentialJsonSchemaSerDesV1.scala

```diff
@@ -0,0 +1,234 @@
+package io.iohk.atala.pollux.core.model.schema.`type`
+
+import io.iohk.atala.pollux.core.model.schema.validator.SchemaSerDes
+import zio.*
+import zio.json.*
+import zio.json.ast.Json
+
+case class CredentialJsonSchemaSerDesV1(
+    $schema: String,
+    $id: Option[String],
+    description: Option[String],
+    properties: Json,
+    required: Set[String],
+    additionalProperties: Option[Boolean]
+)
+
+object CredentialJsonSchemaSerDesV1 {
+  val version: String = "CredentialSchemaSchemaV1"
+
+  private val schema: String = """
+      |{
+      |  "$schema": "https://json-schema.org/draft/2020-12/schema",
+      |  "$id": "https://example.com/custom-meta-schema",
+      |  "type": "object",
+      |  "properties": {
+      |    "$id": {
+      |      "type": "string",
+      |      "format": "uri-reference"
+      |    },
+      |    "$schema": {
+      |      "type": "string",
+      |      "format": "uri",
+      |      "enum": ["https://json-schema.org/draft/2020-12/schema"]
+      |    },
+      |    "description": {
+      |      "type": "string",
+      |      "minLength": 1,
+      |      "maxLength": 2000
+      |    },
+      |    "type": {
+      |      "enum": ["object"]
+      |    },
+      |    "properties": {
+      |      "type": "object",
+      |      "patternProperties": {
+      |        ".*": {
+      |          "$ref": "#/$defs/oneOf"
+      |        }
+      |      }
+      |    },
+      |    "required": {
+      |      "type": "array",
+      |      "items": {
+      |        "type": "string",
+      |        "minLength": 1
+      |      },
+      |      "minItems": 0,
+      |      "maxItems": 125,
+      |      "uniqueItems": true
+      |    },
+      |    "additionalProperties": {
+      |      "type": "boolean"
+      |    }
+      |  },
+      |  "required": ["$schema","type","properties"],
+      |  "additionalProperties": false,
+      |  "$defs": {
+      |    "oneOf": {
```

Comment:
Why do we need two `oneOf` terms?

Reply:
Well pointed out @yshyn-iohk. The top-level one is actually just the name/id for the inner `oneOf` def declaration. I will indeed give it another name (i.e. `oneOfDef`) to avoid any confusion 👍